aws/data_source_aws_ram_resource_share_test.go

```diff
@@ -0,0 +1,99 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccDataSourceAwsRamResourceShare_Basic(t *testing.T) {
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_ram_resource_share.test"
+	datasourceName := "data.aws_ram_resource_share.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccDataSourceAwsRamResourceShareConfig_NonExistent,
+				ExpectError: regexp.MustCompile(`No matching resource found`),
+			},
+			{
+				Config: testAccDataSourceAwsRamResourceShareConfig_Name(rName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair(datasourceName, "name", resourceName, "name"),
+					resource.TestCheckResourceAttrPair(datasourceName, "id", resourceName, "id"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAwsRamResourceShare_Tags(t *testing.T) {
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_ram_resource_share.test"
+	datasourceName := "data.aws_ram_resource_share.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAwsRamResourceShareConfig_Tags(rName),
+				Check: resource.ComposeTestCheckFunc(
```

Comment:
This test should be explicitly checking the tags matches the resource, at least by count, e.g.

```go
resource.TestCheckResourceAttrPair(datasourceName, "tags.%", resourceName, "tags.%"),
```